operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -697,18 +747,115 @@ private void addIncludesSearchParameters(String compartmentMemberType, Multivalu
+            for (Reference reference: references) {
+                ReferenceValue referenceValue = ReferenceUtil.createReferenceValueFrom(reference, compartmentMemberType);
+                if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_RELATIVE)) {
+                    String externalId = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
+                    String[] splitString = externalId.split("/");
+                    String type = splitString[0];
+                    if (!resourceIds.contains(externalId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        resourceList.add(resourceHelper.doRead(type, splitString[1], false, false, null, null).getResource());
+                        resourceIds.add(externalId);
+                    }
+                } else if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_ABSOLUTE)) {
+                    String externalUrl = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
+                    int lastIndexOf = externalUrl.lastIndexOf("/");
+                    String externalId = externalUrl.substring(lastIndexOf+1, externalUrl.length());
+                    String tempString = externalUrl.substring(0,lastIndexOf);
+                    lastIndexOf = tempString.lastIndexOf("/");
+                    String type = tempString.substring(lastIndexOf+1, tempString.length());
+                    if (!resourceIds.contains(externalId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        resourceList.add(resourceHelper.doRead(type, externalId, false, false, null, null).getResource());
+                        resourceIds.add(type + "/" + externalId);
+                    }
+                }
+            }
+        }
+
+        // Transform resources into Entries
+        for (Resource resource: resourceList) {
+            // Hit a case where if a reference points to an object that doesn't exist, 
+            // i.e.- "asserter":{"reference":"Practitioner/DollarEverythingTest-DOESNOTEXIST"},
+            // the resource object is null
+            if (resource != null) {
+                Bundle.Entry.Builder entryBuilder = Bundle.Entry.builder();
+                
+                entryBuilder.fullUrl(Uri.of(url + "/" + resource.getClass().getSimpleName() + "/" + resource.getId()));
+                entryBuilder.resource(resource);
```

Comment:
food for thought:  if we did the read above via a bundle of reads (using FHIRRestHelper.doBundle instead of FHIRRestHelper.doRead on each one), then we'd get them back as bundle entries and we wouldn't need to wrap them

Reply:
Done.